src/common/ceph_argparse.cc

```diff
@@ -521,22 +521,23 @@ CephInitParameters ceph_argparse_early_args
 
 static void generic_usage(bool is_server)
 {
-  cout << "\
-  --conf/-c FILE    read configuration from the given configuration file\n\
-  --id/-i ID        set ID portion of my name\n\
-  --name/-n TYPE.ID set name\n\
-  --cluster NAME    set cluster name (default: ceph)\n\
-  --setuser USER    set uid to user or uid (and gid to user's gid)\n\
-  --setgroup GROUP  set gid to group or gid\n\
-  --version         show version and quit\n\
-" << std::endl;
+  cout << std::endl <<
+    "--conf/-c FILE    read configuration from the given configuration file" << std::endl <<
+    (is_server ?
+    "--id/-i ID        set ID portion of my name" :
+    "--id ID           set ID portion of my name") << std::endl <<
+    "--name/-n TYPE.ID set name" << std::endl <<
+    "--cluster NAME    set cluster name (default: ceph)" << std::endl <<
+    "--setuser USER    set uid to user or uid (and gid to user's gid)" << std::endl <<
+    "--setgroup GROUP  set gid to group or gid" << std::endl <<
+    "--version         show version and quit" << std::endl;
```

Comment:
See [this](https://jenkins.ceph.com/job/ceph-pull-requests/41459/consoleFull#259433645609c5361-bd50-4947-89df-91e1b6c97aa1). It still shows the entire `cout` statement as a diff. Is it because your change (std::endl) has removed the 2 character indentation for each option line ? Because that is the only difference in the jenkins diff. Also I see a newline is missing at the end when `is_server == false`.

(This may sound silly, but I remember I have faced similar build issue and I have made the usage and help.t exactly the same when I have updated rgw_admin.cc options, in-order to get a successful build).

The help.t changes seem to be good, as only 1 line has changed and it was working fine prior to the change.

Reply:
Okay,  I think I resolved the issues. Thanks Jos!